deployments/tekton.go

```diff
@@ -47,145 +47,206 @@ func (k *Tekton) Restore(c *kubernetes.Cluster, ui *ui.UI, d string) error {
+					return "", c.DeleteNamespace(ns)
+				},
+			)
+			if err != nil {
+				return errors.Wrapf(err, "Failed deleting namespace %s", ns)
+			}
+			wasDeleted = true
+		}
 	}
 
-	message := "Deleting Tekton namespace " + tektonNamespace
-	_, err = helpers.WaitForCommandCompletion(ui, message,
-		func() (string, error) {
-			return "", c.DeleteNamespace(tektonNamespace)
-		},
-	)
-	if err != nil {
-		return errors.Wrapf(err, "Failed deleting namespace %s", tektonNamespace)
+	if wasDeleted {
+		ui.Success().Msg("Tekton removed")
 	}
 
-	ui.Success().Msg("Tekton removed")
-
 	return nil
 }
 
 func (k Tekton) apply(c *kubernetes.Cluster, ui *ui.UI, options kubernetes.InstallationOptions, upgrade bool) error {
-	if out, err := helpers.KubectlApplyEmbeddedYaml(tektonAdminRoleYamlPath); err != nil {
-		return errors.Wrap(err, fmt.Sprintf("Installing %s failed:\n%s", tektonAdminRoleYamlPath, out))
-	}
-	if out, err := helpers.KubectlApplyEmbeddedYaml(tektonPipelineYamlPath); err != nil {
-		return errors.Wrap(err, fmt.Sprintf("Installing %s failed:\n%s", tektonPipelineYamlPath, out))
-	}
-	if out, err := helpers.KubectlApplyEmbeddedYaml(tektonTriggersYamlPath); err != nil {
-		return errors.Wrap(err, fmt.Sprintf("Installing %s failed:\n%s", tektonTriggersYamlPath, out))
-	}
-	if out, err := helpers.KubectlApplyEmbeddedYaml(tektonDashboardYamlPath); err != nil {
-		return errors.Wrap(err, fmt.Sprintf("Installing %s failed:\n%s", tektonDashboardYamlPath, out))
-	}
 
-	err := c.LabelNamespace(tektonNamespace, kubernetes.FusemlDeploymentLabelKey, kubernetes.FusemlDeploymentLabelValue)
-	if err != nil {
-		return err
+	installOperator := true
+	if !upgrade {
+		// Install tekton operator only if not already installed
+		_, err := c.Kubectl.CoreV1().Namespaces().Get(
+			context.Background(),
+			tektonOperatorNamespace,
+			metav1.GetOptions{},
+		)
+		if err == nil {
+			ui.Exclamation().Msg("Tekton Operator already present, skipping its installation")
+			installOperator = false
+		}
 	}
 
-	for _, crd := range []string{
-		"clustertasks.tekton.dev",
-		"clustertriggerbindings.triggers.tekton.dev",
-		"conditions.tekton.dev",
-		"eventlisteners.triggers.tekton.dev",
-		"pipelineresources.tekton.dev",
-		"pipelineruns.tekton.dev",
-		"pipelines.tekton.dev",
-		"runs.tekton.dev",
-		"taskruns.tekton.dev",
-		"tasks.tekton.dev",
-		"triggerbindings.triggers.tekton.dev",
-		"triggers.triggers.tekton.dev",
-		"triggertemplates.triggers.tekton.dev",
-	} {
-		message := fmt.Sprintf("Establish CRD %s", crd)
-		out, err := helpers.WaitForCommandCompletion(ui, message,
-			func() (string, error) {
-				return helpers.Kubectl("wait --for=condition=established --timeout=" + strconv.Itoa(k.Timeout) + "s crd/" + crd)
-			},
-		)
+	if installOperator || upgrade {
+		if out, err := helpers.KubectlApplyEmbeddedYaml(tektonOperatorYamlPath); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("installing %s failed:\n%s", tektonOperatorProfileYamlPath, out))
```

Comment:
I think `tektonOperatorProfileYamlPath` should be `tektonOperatorYamlPath`

Reply:
yep, good catch!